CNV-65873: Allow users to enable the cross cluster live migration feature (#3654)
kubevirt/hyperconverged-cluster-operator: docs/cluster-configuration.md

To enable a feature, add its name to the `featureGates` list and set it to `true`. Missing or `false` feature gates
disables the feature.

> Some of the feature gates are used to set feature gates in the KubeVirt CR. This feature gates are described in the

Author: Canceled the refactoring of the doc, and removed this text.

> Some of the feature gates are used to set feature gates in the KubeVirt CR. This feature gates are described in the

Author: Dropped this text

[KubeVirt feature](#kubevirt-feature-gates) gates sections.

### withHostPassthroughCPU Feature Gate
This feature gate is deprecated and is ignored.

### deployKubeSecondaryDNS Feature Gate
Set the `deployKubeSecondaryDNS` feature gate to true to allow deploying KubeSecondaryDNS by CNAO.
For additional information, see here: [KubeSecondaryDNS](https://github.com/kubevirt/kubesecondarydns)

**Default**: `false`

### Feature Gates Example

```yaml
apiVersion: hco.kubevirt.io/v1beta1
kind: HyperConverged
metadata:
  name: kubevirt-hyperconverged
spec:
  infra: {}
  workloads: {}
  featureGates:
    alignCPUs: true
    deployKubeSecondaryDNS: true
```

## KubeVirt feature gates

> The following feature gates are used to set feature gates in the KubeVirt CR. For few of them, HCO will perform some

Reviewer: same as before

> more functionality, beside just setting the feature gate in the KubeVirt CR.

Reviewer: `some more functionality` is a bit cryptic

Author: Removed

### downwardMetrics Feature Gate
Set the `downwardMetrics` feature gate in order to allow exposing a limited set of VM and host metrics to the guest.
The format is compatible with [vhostmd](https://github.com/vhostmd/vhostmd).
These metrics allow third-parties diagnosing issues.
DownwardMetrics may be exposed to the guest through a `volume` or a `virtio-serial port`.

> **Note**: Updates from previous versions require to enable this feature gate if the metrics was used in any

Reviewer: ```suggestion
**Note**: Updates from previous versions require enabling this feature gate if the metrics were used in any
```
